### PR TITLE
adding poll creation option

### PR DESCRIPTION
**Pull request by ingawei**
- added the ability to create a poll contract in dev
image
- for a poll, default close time (when you can stop voting and a majority is automatically chosen) is never (this never option is only for polls and is actually just 1000 years in the future)
image
- pretty short PR but I'd like to get your feedback on implementation so far and also a second set of eyes because I'm feeling a the kind of sick where you're losing your mind so this code might be gross

### CODE REVIEW

File: backend/api/src/create-market.ts [hunk 1]
```diff
@@ -441,7 +442,8 @@ async function generateAntes(
     outcomeType === 'BINARY' ||
     outcomeType === 'PSEUDO_NUMERIC' ||
     outcomeType === 'STONK' ||
-    outcomeType === 'MULTIPLE_CHOICE'
+    outcomeType === 'MULTIPLE_CHOICE' ||
+    outcomeType === 'POLL'
```
Comment (jahooma) [inline backend/api/src/create-market.ts:446]: Generating a liquidity doc for polls? We can skip all this for polls.

File: web/components/new-contract/contract-params-form.tsx [hunk 2]
```diff
@@ -154,6 +155,19 @@ export function ContractParamsForm(props: {
...
     : undefined
 
+  const closeDateMap: { [key: string]: number | string } = {
+    'A day': 1,
+    'A week': 7,
+    '30 days': 30,
+    'This year': daysLeftInTheYear,
+  }
+
+  const NEVER_IN_DAYS = 30 * 12 * 1000 // ~1000 years
```
Comment (jahooma) [inline web/components/new-contract/contract-params-form.tsx:165]: I think Twitter polls always close (seemingly too soon — I feel like I often still want to vote in them).

I'm ok trying out this Never option. We'll see if users want it.

File: web/components/new-contract/contract-params-form.tsx [hunk 3]
```diff
@@ -154,6 +155,19 @@ export function ContractParamsForm(props: {
...
+  if (outcomeType == 'POLL') {
+    closeDateMap['Never'] = NEVER_IN_DAYS
+  }
+  const [neverCloses, setNeverCloses] = useState(false)
```
Comment (jahooma) [inline web/components/new-contract/contract-params-form.tsx:169]: Does this need to be a separate useState? Isn't it just:

`const neverCloses = closeDateInDays === NEVER_IN_DAYS`

**Review by jahooma — APPROVED** [review]
Looks pretty good, Inga! I think the data schema is right. I guess there's one more table to be created for recording user votes in the poll.

File: web/components/new-contract/contract-params-form.tsx (hunk 3 above)
Comment (ingawei) [reply web/components/new-contract/contract-params-form.tsx:169]: closeDateInDays doesn't exist, it just adds that amount to now

**Pull request merged**